app/controllers/api/v1/users_controller.rb

```diff
@@ -1,48 +1,34 @@
 class Api::V1::UsersController < ApplicationController
-  before_action :set_user, only: %i[show update delete]
-
-  def index
-    @users = User.all
-    render json: @users if @users
-  end
-
-  def show
-    render json: @user if @user
-  end
-
+  # REGISTER
   def create
     @user = User.create(user_params)
-
-    if @user.save
-      render json: { success: true, message: 'User loged in successfully.' }, status: :created
+    if @user.valid?
+      # token = encode_token({user_id: @user.id})
+      render json: { user: @user }
     else
-      render json: { success: false, message: @user.error.full_messages }, status: :unprocessable_entity
+      render json: { error: 'Invalid username or password' }
     end
   end
 
-  def update
-    if @user.update(user_params)
-      render json: { success: true, message: 'User updated successfully', user: @user }
+  # LOGGING IN
+  def login
+    @user = User.find_by(username: params[:username])
```

Comment:
- [ ] We don't use username to select the user uniquely so change it with email.

```ruby
@user = User.find_by(email: params[:email])
```